Fundamentals/Backoffice/Property-Editors/Built-in-Property-Editors/Block-Editor/build-custom-view-for-blocks.md

```diff
@@ -0,0 +1,133 @@
+</button>
+```
+
+If you would like to display properties of `settings`, you can access these by using `block.settingsData.myPropertyAlias`.
+
+## Add the Areas Container for Block Grid Editor
+
+Blocks of the Block Grid Editor can have Areas. These Blocks requires their Custom View to define where the Area Container should be inserted.
+
+This enables you to wrap and place the Areas as you need.
+
+The Area Container is inserted through a slot. Insert the following snippet where you would like the Area to appear in your Custom View:
+
+```html
+<slot name="areas" part="areas"></slot>
+```
+
+## Adding custom implementation to your View
+
+To achieve this you need to add a custom AngularJS controller to your custom view, using the `ng-controller` attribute:
+
+```html
+<button type="button" ng-controller="customBlockController" ng-click="api.editBlock(block, block.hideContentInOverlay, index, parentForm)" class="btn-reset umb-outline blockelement-labelblock-editor blockelement__draggable-element ng-scope ui-sortable-handle" >
+    <h2 ng-bind="block.data.headline"></h2>
+    <p ng-bind="block.data.description"></p>
+</button>
+```
+
+:::note
+The class declaration is necessary to allow the block to behave like the default blocks, like the "drag-to-sort" feature.
+:::
+
+Create a folder inside the `App_Plugins` folder called 'CustomBlockView'.
+
+Create two files within the CustomBlockView file: `package.manifest` and `customBlock.controller.js`.
+ 
+Add the following JSON to the `package.manifest` file:
+
+```json
+{
+  "javascript": [
+    "~/App_Plugins/CustomBlockView/customBlock.controller.js"    
+  ]  
+}
+```
+
+:::note
+Umbraco will parse all `package.manifest` files and load any resources they reference into the backoffice during startup.
+:::
+
+The second file, `customBlock.controller.js`, will be used to register the 'customBlockController' defined using the `ng-controller` attribute in your custom view.
+
+To register the controller, add the following lines of code:
+
+```javascript
+angular.module("umbraco").controller("customBlockController", function ($scope) {
+// you can do your custom functionality here!
+});
+```
+
+### Example: Displaying an image from a Media Picker
+
+Your block may enable you to 'pick' an image for use as the background for a particular block. If you try to display this image directly in the view using `block.data.image`, you will see the unique ID and not the image.
+
+You will need to use the ID in our custom AngularJS controller in order to get the `ImageUrl` to display in our backoffice Block Editor View.
+
+With the setup of files above, you need to amend the `customBlock.controller.js` file, by injecting the `mediaResource` to retrieve the image from the ID:
+
+```javascript
+angular.module("umbraco").controller("customBlockController", function ($scope, mediaResource) {
+
+    //your property is called image so the following will contain the udi:
+        var imageUdi = $scope.block.data.image;
```

Comment:
The OCD in me @nielslyngsoe is that the tabs for this are inconsistent.

Reply:
Oh yes, corrected